app/services/work_history_with_breaks.rb

```diff
@@ -0,0 +1,80 @@
+class WorkHistoryWithBreaks
+  class BreakPlaceholder
+    def initialize(month_range:)
+      @month_range = month_range
+    end
+
+    def start_date
+      @month_range.first
+    end
+
+    def length
+      @month_range.count
+    end
+  end
+
+  def initialize(work_history)
+    @work_history = work_history.sort_by(&:start_date)
+    @work_history_with_breaks = []
+    @current_job = nil
+  end
+
+  def call
```

Comment:
This method name could be something like `timeline` which might help make the included code a bit more readable